cura/Arranging/GridArrange.py

```diff
@@ -0,0 +1,267 @@
+import math
+from typing import List, TYPE_CHECKING, Optional, Tuple, Set
+
+from UM.Application import Application
+from UM.Math import AxisAlignedBox
+from UM.Math.Vector import Vector
+from UM.Operations.AddSceneNodeOperation import AddSceneNodeOperation
+from UM.Operations.GroupedOperation import GroupedOperation
+from UM.Operations.TranslateOperation import TranslateOperation
+
+
+class GridArrange:
+    offset_x: float = 10
+    offset_y: float = 10
+
+    _grid_width: float
+    _grid_height: float
+
+    _nodes_to_arrange: List["SceneNode"]
+    _fixed_nodes: List["SceneNode"]
+    _build_volume: "BuildVolume"
+    _build_volume_bounding_box: AxisAlignedBox
+
+    def __init__(self, nodes_to_arrange: List["SceneNode"], build_volume: "BuildVolume", fixed_nodes: List["SceneNode"] = []):
```

Comment:
```suggestion
    def __init__(self, nodes_to_arrange: List["SceneNode"], build_volume: "BuildVolume", fixed_nodes: List["SceneNode"] = []):
```

You shouldn't use `fixed_nodes: List["SceneNode"] = []` Python is really weird in that sense. See https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments for a bit more info about it.

Reply:
fixed in https://github.com/Ultimaker/Cura/pull/15496/commits/118f49a0520bdd1a3983bbf33b49c3ed8a645fb1